### ISSUE TITLE
Don't see event counters with own TraceEventSession but PerfView does see them

### ISSUE THREAD

**Issue by BruceCowan-AI** [Microsoft/perfview#801]
I am trying to pick up event counters in my own TraceEventSession, but I don't get them. PerfView, run at the same time with the same collect arguments, does see them. I am using .NET Framework 4.7.2. What am I doing wrong?

My code to record the events looks like this:
`_traceSession = new TraceEventSession("AquariusTestTraceSession");
            _traceSession.EnableProvider(TplEtwProviderTraceEventParser.ProviderGuid, eventLevel, (ulong)TplEtwProviderTraceEventParser.Keywords.TasksFlowActivityIds);
            _traceSession.EnableProvider(EtwNames.TimeSeriesProviderName, eventLevel);
            var options = new TraceEventProviderOptions("EventCounterIntervalSec", eventCounterIntervalSec.ToString());
            _traceSession.EnableProvider(EtwNames.ProcessingProviderName, eventLevel, options: options);
            var testAllEventStream = _traceSession.Source.ObserveAll();
            testAllEventStream.Subscribe(ev =>
            {
                Console.WriteLine($"All received {ev.EventName} activityId {ActivityPathString(ev.ActivityID)} from {ev.ProviderName}");
            });`

**Comment by vancem** [comment]
Hmm, I can't see anything obvious.  If can provide a minimal repro (a simple sample that has your EventSource with its EventCounters) as a ZIPed solution I can give you more useful advice.

**Comment by BruceCowan-AI** [comment]
Thanks very much for the offer. I made a simple sample and it is even worse - it sees no events at all, but the PerfView running at the same time sees all the events including the counters. I can't see what I did wrong.
[EventListenDemo.zip](https://github.com/Microsoft/perfview/files/2616887/EventListenDemo.zip)

**Comment by adamsitnik** [comment]
@BruceCowan-AI it looks like you are missing `_traceSession.Source.Process();` at the end

you can use [this](https://github.com/Microsoft/perfview/blob/master/src/TraceEvent/Samples/22_ObserveEventSource.cs) sample to also add support for Ctrl+C and session disposal

**Comment by BruceCowan-AI** [comment]
Thanks @adamsitnik, that is indeed why my sample did not work. And with the fix the sample does see the event counters, so I'm left with the mystery of why my test code does not (as I said above, my test code sees all the events except the counters). So a puzzle for me to solve.

**Comment by briangithex** [comment]
It's probably too late for BruceCowan to care, but I encountered the same problem.  So, I'll post the answer for anyone who runs the same search I did and finds this thread:

Typically, perfview only sees counters if you add "EventCounterIntervalSec=1" as part of the commandline.  To do the same thing in with a `TraceEventProvider`, use code similar to this:

Something like this
```
var options = new TraceEventProviderOptions();
options.AddArgument("EventCounterIntervalSec", "1");
session.EnableProvider(providerName,options:options);
```

If my version of this fails to compile, Vancem's version [here](https://github.com/dotnet/runtime/issues/21983#issuecomment-308169216), which uses slightly different classes, might work.  Either way, knowing to use the options parameter when calling `EnableProvider` is hopefully enough to unstick most people with this issue.
